indico/modules/categories/controllers/display.py

```diff
@@ -567,22 +581,33 @@ def _process(self):
 
 
 class RHCategoryCalendarViewEvents(RHDisplayCategoryBase):
-    def _process_args(self):
+    class GroupBy(Enum):
+        category = auto()
+        location = auto()
+
+    @use_kwargs({'group_by': EnumField(GroupBy, load_default='category')}, location='query')
+    def _process_args(self, group_by):
         RHDisplayCategoryBase._process_args(self)
         tz = self.category.display_tzinfo
+        self.group_by = group_by
         try:
             self.start_dt = tz.localize(dateutil.parser.parse(request.args['start'])).astimezone(utc)
             self.end_dt = tz.localize(dateutil.parser.parse(request.args['end'])).astimezone(utc)
         except ParserError as e:
             raise BadRequest(str(e))
 
     def _process(self):
-        query = (Event.query
+        query = (
+            Event.query
```

Comment:
Please keep the formatting style we use everywhere else in Indico.

That would be one of these:

```python
        query = (
            Event.query
            .filter(Event.starts_between(self.start_dt, self.end_dt),
                    Event.is_visible_in(self.category.id),
                    ~Event.is_deleted)
            .options(undefer(Event.detailed_category_chain),
                     load_only('id', 'title', 'start_dt', 'end_dt', 'category_id', 'own_venue_id'))
        )
```

```python
        query = (Event.query
                 .filter(Event.starts_between(self.start_dt, self.end_dt),
                         Event.is_visible_in(self.category.id),
                         ~Event.is_deleted)
                 .options(undefer(Event.detailed_category_chain),
                          load_only('id', 'title', 'start_dt', 'end_dt', 'category_id', 'own_venue_id')))
```

I think the second version is more common (and in fact it's what the `ongoing_events` query right below uses).